src/core.lisp

```diff
@@ -70,19 +75,18 @@
                             :initial-element :infinity)))
     (dolist (triangle (scene:scene-triangles scene))
       (let* ((flat-triangle (project-triangle-to-camera camera triangle))
-             (minimal-x (max 0 (reduce #'min flat-triangle :key #'d-p-x)))
+             (minimal-x (max 0 (floor (reduce #'min flat-triangle :key #'d-p-x))))
              (maximal-x (min (1- (scene:camera-width camera))
-                             (reduce #'max flat-triangle :key #'d-p-x)))
-             (minimal-y (max 0 (reduce #'min flat-triangle :key #'d-p-y)))
+                             (ceiling (reduce #'max flat-triangle :key #'d-p-x))))
+             (minimal-y (max 0 (floor (reduce #'min flat-triangle :key #'d-p-y))))
              (maximal-y (min (1- (scene:camera-height camera))
-                             (reduce #'max flat-triangle :key #'d-p-y))))
+                             (ceiling (reduce #'max flat-triangle :key #'d-p-y)))))
         (loop for pixel-x from minimal-x to maximal-x
```

Comment:
Maybe we should use centers of the pixels? And iterate from minimal-x + 1/2?
Same for y.

Reply:
I added additional variables `point-x` and `point-y` which are set to `pixel-x/y + 1/2`. I kept the original values because they are used as array indices.